rs/ethereum/cketh/minter/src/eth_logs/parser.rs

```diff
@@ -164,22 +164,29 @@ impl LogParser for ReceivedEthOrErc20LogParser {
 fn ensure_not_pending(entry: &LogEntry) -> Result<(BlockNumber, EventSource), ReceivedEventError> {
     let _block_hash = entry
         .block_hash
+        .clone()
         .ok_or(ReceivedEventError::PendingLogEntry)?;
     let block_number = entry
         .block_number
+        .clone()
         .ok_or(ReceivedEventError::PendingLogEntry)?;
     let transaction_hash = entry
         .transaction_hash
+        .clone()
         .ok_or(ReceivedEventError::PendingLogEntry)?;
     let _transaction_index = entry
         .transaction_index
+        .clone()
+        .ok_or(ReceivedEventError::PendingLogEntry)?;
+    let log_index = entry
+        .log_index
+        .clone()
         .ok_or(ReceivedEventError::PendingLogEntry)?;
-    let log_index = entry.log_index.ok_or(ReceivedEventError::PendingLogEntry)?;
     Ok((
-        block_number,
+        block_number.into(),
         EventSource {
-            transaction_hash,
-            log_index,
+            transaction_hash: Hash(<[u8; 32]>::try_from(transaction_hash.as_ref()).unwrap()),
```

Comment:
Note that `transaction_hash` is a `Hex32`, which as convertions to `[u8; 32]` so this should work
```suggestion
            transaction_hash: Hash(transaction_hash.into()),
```

Reply:
Nice, thanks!